domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public void update(AsyncResult<SQLConnection> sqlConnection, String table, Object entity, String jsonbField,
                      String whereClause, boolean returnUpdatedIds, Handler<AsyncResult<RowSet<Row>>> replyHandler) {
-    if (conn.failed()) {
-      replyHandler.handle(Future.failedFuture(conn.cause()));
-      return;
-    }
-    long start = System.nanoTime();
-    StringBuilder sb = new StringBuilder();
-    sb.append(whereClause);
-    StringBuilder returning = new StringBuilder();
-    if (returnUpdatedIds) {
-      returning.append(RETURNING_ID);
-    }
-    try {
-      String q = UPDATE + schemaName + DOT + table + SET + jsonbField + " = $1::jsonb " + whereClause
-          + SPACE + returning;
-      log.debug("update query = " + q);
-      conn.result().conn.preparedQuery(q).execute(Tuple.of(pojo2JsonObject(entity)), query -> {
-        if (query.failed()) {
-          log.error(query.cause().getMessage(), query.cause());
-        }
-        statsTracker(UPDATE_STAT_METHOD, table, start);
-        replyHandler.handle(query);
-      });
-    } catch (Exception e) {
-      log.error(e.getMessage(), e);
-      replyHandler.handle(Future.failedFuture(e));
-    }
+
+    withConn(sqlConnection, conn -> conn.update(table, entity, jsonbField, whereClause, returnUpdatedIds))
+    .onComplete(replyHandler);
+  }
+
+  /**
+   * Update all records in {@code table} that match the SQL {@code whereClause} query.
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @return one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, String jsonbField, String whereClause, boolean returnUpdatedIds) {
+    return withConn(conn -> conn.update(table, entity, jsonbField, whereClause, returnUpdatedIds));
   }
 
+  /**
+   * Update all records in {@code table} that match the SQL {@code whereClause} query.
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
   public void update(String table, Object entity, String jsonbField, String whereClause, boolean returnUpdatedIds,
```

Comment:
Javadoc

Reply:
`@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment